core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1119,6 +1127,15 @@ class ReplicaManager(val config: KafkaConfig,
         else
           Set.empty[Partition]
 
+        // iterate through all the partitions which the broker is a leader of
+        // and remove any topic that is found since the current broker still needs to update its metrics
+        // for those topics
+        // in all of the partitions that the current broker is now a follower of
+        // add all the topics in a list as candidate to remove metrics
+        // and finally remove all the metrics of topics that the current broker IS NO LONGER a leader of
+        val topicNotToInclude = leaderPartitionsIterator.map(_.topic).toSet
```

Comment:
nit: name `topicNotToInclude` is a bit confusing, how about just `leaderTopicSet`?